src/neptune_tensorboard/integration/tensorflow_integration.py

```diff
@@ -13,255 +13,93 @@
-
-    tf.summary.image = image
-    tf.summary._original_no_neptune_image = _image
-
-    tf.summary.text = text
-    tf.summary._original_no_neptune_text = _text
-
-    # Tensorflow 2.3 renames the internal method from `_log_metrics` to `_log_epoch_metrics`
-    # and changes its parameters. The conditional below handles both versions.
-    if hasattr(tf.keras.callbacks.TensorBoard, "_log_metrics"):
-
-        _tb_log_metrics = tf.keras.callbacks.TensorBoard._log_metrics
-
-        def _log_metrics(instance, logs, prefix, step):
-            exp = experiment_getter()
-            for (name, value) in logs.items():
-                if name in ("batch", "size", "num_steps"):
-                    continue
-                exp.log_metric(get_channel_name(name), x=step, y=value)
-
-            _tb_log_metrics(instance, logs, prefix, step)
-
-        tf.keras.callbacks.TensorBoard._log_metrics = _log_metrics
-
-    elif hasattr(tf.keras.callbacks.TensorBoard, "_log_epoch_metrics"):
-
-        _tb_log_epoch_metrics = tf.keras.callbacks.TensorBoard._log_epoch_metrics
+def track_image(name, data, step=None, run=None, base_namespace=None):
+    # expecting 2 or 3 dimensional tensor. If tensor is 4-dimentional,
+    # as in https://www.tensorflow.org/api_docs/python/tf/summary/image
+    # iterate over first dimension to send all images
+    shape = tf.shape(data)
+    if len(shape) >= 4:
+        for num in range(0, shape[0]):
+            run[base_namespace]["image"][name].append(File.as_image(data[num]))
+    else:
+        run[base_namespace]["image"][name] = File.as_image(data)
 
-        def _log_epoch_metrics(instance, epoch, logs):
-            exp = experiment_getter()
-            for (name, value) in logs.items():
-                if name in ("batch", "size", "num_steps"):
-                    continue
-                exp.log_metric(get_channel_name(name), x=epoch, y=value)
 
-            _tb_log_epoch_metrics(instance, epoch, logs)
+def track_text(name, data, step=None, description=None, run=None, base_namespace=None):
+    run[base_namespace]["text"][name] = data
 
-        tf.keras.callbacks.TensorBoard._log_epoch_metrics = _log_epoch_metrics
 
+def track_graph(graph_data, run=None, base_namespace=None):
+    if IS_GRAPHLIB_AVAILABLE:
+        graph = tfg.board(graph_data)
+        png_bytes = graph.pipe(format="png")
+        # There is only one graph
+        run[base_namespace]["graph"].upload(File.from_content(png_bytes, extension="png"))
     else:
-        raise AttributeError("Unsupported Tensorflow version")
+        warnings.warn("Skipping model visualization because no tfgraphviz installation was found.")
+
+
+def register_pre_hook(original, neptune_hook, run, base_namespace):
+    @wraps(original)
+    def wrapper(*args, **kwargs):
+        neptune_hook(*args, **kwargs, run=run, base_namespace=base_namespace)
+        return original(*args, **kwargs)
+
+    return wrapper
+
+
+def _patch_tensorflow_2x(run, base_namespace):
```

Comment:
Let's use `__all__` instead of an underscore as a prefix :wink:. This will be enough to indicate that this function is private across module.